modules/client/src/test/kotlin/jp/co/soramitsu/iroha2/TriggersTest.kt

```diff
@@ -248,49 +254,75 @@ class TriggersTest : IrohaTest<Iroha2Client>() {
                 Metadata(mapOf()),
                 filter,
             )
-            buildSigned(ALICE_KEYPAIR)
-        }.also { d ->
-            withTimeout(txTimeout) { d.await() }
         }
 
-        // send some transactions to keep Iroha2 network busy
-        repeat(2) { i ->
-            client.sendTransaction {
-                accountId = ALICE_ACCOUNT_ID
-                setKeyValue(ALICE_ACCOUNT_ID, "test$i".asName(), "test$i".asValue())
-                buildSigned(ALICE_KEYPAIR)
-            }.also { d ->
-                withTimeout(txTimeout) { d.await() }
-            }
-        }
-        QueryBuilder.findAssetsByAccountId(ALICE_ACCOUNT_ID)
+        keepNetworkBusyAndCheckAssetDefinitionIds()
+
+        val testKey = "key"
+        val testValue = "value"
+        client.tx { setKeyValue(triggerId, testKey.asName(), testValue.asValue()) }
+        QueryBuilder.findTriggerById(triggerId)
             .account(ALICE_ACCOUNT_ID)
             .buildSigned(ALICE_KEYPAIR)
             .let { query -> client.sendQuery(query) }
-            .also { assets ->
-                assert(assets.size > 1)
-                assert(assets.all { it.id.accountId == ALICE_ACCOUNT_ID })
-                assert(assets.any { it.id.definitionId == XOR_DEFINITION_ID })
-                assert(
-                    assets.any {
-                        it.id.definitionId == AssetDefinitionId(
-                            "nft_number_1_for_alice".asName(),
-                            DEFAULT_DOMAIN_ID,
-                        )
-                    },
-                )
-            }
+            .also { assertEquals(testValue, it.action.metadata.getStringValue(testKey)) }
+    }
+
+    @Test
+    @WithIroha([AliceHas100XorAndPermissionToBurn::class])
+    @Story("Wasm trigger mints NFT for every user when trigger metadata is updated")
+    @SdkTestId("wasm_trigger_to_mint_nft_for_every_user_on_update_trigger_metadata_event")
+    fun `wasm trigger to mint nft for every user on update trigger metadata event`(): Unit = runBlocking {
+        val wasmTriggerId = TriggerId(name = "wasm_trigger".asName())
+        val setKeyValueTriggerId = TriggerId(name = "update_trigger".asName())
+
+        val filter = Filters.data(
+            DataEntityFilter.ByTrigger(
```

Comment:
The point is not to replace one line, but to make the expression smaller overall. It should look like `Filters.byTrigger(wasmTriggerId, ...)`

Reply:
fixed